kubernetes/eventhandler.go

```diff
@@ -157,21 +167,34 @@ func (n *namespacePodUpdater) OnUpdate(obj interface{}) {
 		return
 	}
 
-	// n.store.List() returns a snapshot at this point. If a delete is received
-	// from the main watcher, this loop may generate an update event after the
-	// delete is processed, leaving configurations that would never be deleted.
-	// Also this loop can miss updates, what could leave outdated configurations.
-	// Avoid these issues by locking the processing of events from the main watcher.
 	if n.locker != nil {
 		n.locker.Lock()
 		defer n.locker.Unlock()
 	}
-	for _, pod := range n.store.List() {
-		pod, ok := pod.(*Pod)
-		if ok && pod.Namespace == ns.Name {
-			n.handler(pod)
+
+	// Slice includes the old and new version of caching object that changes in the current update event. slice[0] is the old version and slice[1] the new updated one
+	slice := n.namespacewatcher.Deltaobjects()
+	cachednamespaceold, ok := slice[0].(*Namespace)
+
+	if ns.Name == cachednamespaceold.Name && ok {
+		labelscheck := checkMetadata(ns.ObjectMeta.Labels, cachednamespaceold.ObjectMeta.Labels)
+		annotationscheck := checkMetadata(ns.ObjectMeta.Annotations, cachednamespaceold.ObjectMeta.Annotations)
+		// Only if there is a diffrence in Metadata labels or annotations proceed to Pod update
+		if !labelscheck || !annotationscheck {
+			// n.store.List() returns a snapshot at this point. If a delete is received
+			// from the main watcher, this loop may generate an update event after the
+			// delete is processed, leaving configurations that would never be deleted.
+			// Also this loop can miss updates, what could leave outdated configurations.
+			// Avoid these issues by locking the processing of events from the main watcher.
```

Comment:
this comment here doesn't make sense anymore. We should either move the comment above to where we lock the object or move that code here instead

Reply:
See comment below